src/components/global/Button.tsx

```diff
@@ -0,0 +1,115 @@
+import { css } from '@emotion/react';
+import styled from '@emotion/styled';
+import THEME from '@src/constants/Colors';
+import WIDTHS from '@src/constants/Widths';
+import { Size, Variant } from '@src/types/Button.type';
+import React, { ButtonHTMLAttributes } from 'react';
+import MoonLoader from 'react-spinners/MoonLoader';
+
+export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
+  variant: Variant;
+  size: Size;
+  rounder: boolean;
+  isLoading?: boolean;
+}
+
+const Button = ({ children, onClick, isLoading, variant, size, rounder }: ButtonProps) => {
+  return (
+    <StyledButton onClick={onClick} variant={variant} size={size} rounder={rounder}>
+      {/* To be fixed */}
+      {isLoading ? <MoonLoader color={THEME.PURPLE300} size={25} /> : children}
+    </StyledButton>
+  );
+};
+
+const ButtonSize = (size: Size) => {
+  if (size === 'Small') {
+    return WIDTHS.BUTTON_SMALL;
+  } else {
+    return WIDTHS.BUTTON_LARGE;
+  }
+};
+
+const ButtonColor = (variant: Variant) => {
+  switch (variant) {
+    case 'Primary':
+      return THEME.PURPLE500;
+    case 'Warning':
+      return THEME.ERROR;
+    case 'Bordered':
+      return THEME.WHITE;
+    case 'Cancel':
+      return THEME.WHITE;
+    case 'Disabled':
+      return THEME.GRAY200;
+    default:
+      const errorCheck: never = variant;
+  }
+};
+
+const ButtonRadius = (rounder: boolean) => {
+  if (rounder) {
+    return WIDTHS.BUTTON_RADIUS_LARGE;
+  } else {
+    return WIDTHS.BUTTON_RADIUS_SMALL;
+  }
```

Comment:
simply하게 바꿀 수 있을 것 같아요!

```suggestion
  if (rounder) {
    return WIDTHS.BUTTON_RADIUS_LARGE;
  }
  
    return WIDTHS.BUTTON_RADIUS_SMALL;
```

Reply:
사실 이걸 고민을 좀 했었어요!! 평소 알고리즘 문제 풀 때는 무조건 적어주신 것 처럼 하는데, 프론트엔드 개발을 할 때, 그니까 숏코딩보다 직관적인 코드가 더 중요할 때에는 if else를 통해서 확실하게 보여주는게 개인적으로는 더 좋은 것 같더라구요! 그래도 리뷰 감사합니다!

Reply:
직관적인 코드를 짜야된다고 해서 무조건 숏코딩을 지양하는 것은 아니에요. 필요한 경우에는 숏코딩이 오히려 더 직관적일 때가 많아요. 코드를 읽었을 때, `if return else return`와 `if return / return` 중 어떤 것이 더 직관적이고, 읽기 편한지 고려해보는 것도 좋아요 :)